filebeat/input/syslog/config.go

```diff
@@ -48,6 +48,7 @@ var defaultConfig = config{
 type syslogTCP struct {
 	tcp.Config    `config:",inline"`
 	LineDelimiter string `config:"line_delimiter" validate:"nonzero"`
+	Framing       string `config:"framing"`
```

Comment:
Could this be make into a `type FramingType` that implements `Validate() error` such that config unpacking will automatically validate the values.

Reply:
tried that using `type FramingType string` and when beats would start I'd get the following panic:
```
instance/beat.go:167    Failed due to panic.    {"panic": "reflect.nameFrom: name too long: *************************************************************************
....
```

Reply:
I think you need to implement https://pkg.go.dev/github.com/elastic/go-ucfg#StringUnpacker. Another option is to make it a `type FramingType uint8` and also implement `Unpack` (that's probably more common in the codebase that having the type be a `string`).

Reply:
Thanks for the help.  Is there any benefit to doing Validate() if Unpack() does validation?

Reply:
Nope. If the type itself has Unpack then no need to add another Validate step with similar logic.